code/game/machinery/vendors/vending.dm

```diff
@@ -799,6 +915,134 @@
 
 /obj/machinery/economy/vending/onTransitZ()
 	return
+
+/obj/machinery/economy/vending/proc/choose_crit()
+	if(!length(possible_crits))
+		return
+	var/crit_path_selected = pick(possible_crits)
+	return GLOB.vendor_crits[crit_path_selected]
+
+/**
+ * Tilts the machine onto the atom passed in.
+ *
+ * Arguments:
+ * * victim - The thing the machine is falling on top of
+ * * crit - if true, some special damage effects might happen.
+ * * from_combat - If true, hold off on some of the additional damage and extra effects.
+ */
+/obj/machinery/economy/vending/proc/tilt(atom/victim, crit = FALSE, from_combat = FALSE)
+	if(QDELETED(src) || !has_gravity(src) || !tiltable || tilted)
+		return
+
+	tilted = TRUE
+	layer = ABOVE_MOB_LAYER
+
+	var/should_throw_at_target = TRUE
+
+	. = FALSE
+
+	if(victim && in_range(victim, src))
+		for(var/mob/living/L in get_turf(victim))
+			var/mob/living/carbon/C = L
+
+			var/crit_rebate = 0
+			var/damage_to_deal = squish_damage
+			if(crit || !from_combat)
```

Comment:
```suggestion
			if(crit && !from_combat)
```
From reading the documentation and the comment below, it looks like it should be this

Reply:
This was actually somewhat intentional. I want it to deal a bit more damage if you knock it over onto yourself but don't crit yourself with it. I'll rework the logic a bit so there's still a factor here, but it isn't as bad as a crit.